docs/how-to/address-ipfs-on-web.md

```diff
@@ -55,49 +55,73 @@ https://ipfs.io/ipns/tr.wikipedia-on-ipfs.org/wiki/Anasayfa.html
 
-Some browsers and other user agents force lowercase for the authority part of URLs, breaking case-sensitive CIDs before HTTP Gateway has a chance to read them.
+Due to these limitaitons, use of short, case-insensitive CIDv1 in subdomain context is advised.
+Base32 is the safe default. Less popular Base36 can be used for longer ED25519 libp2p keys.
 
-To avoid this, use of case-insensitive CIDv1 in Base32 in subdomain context is suggested as the safe default.
+See the next section to learn how to convert existing CID v0 to a DNS-safe representation.
 
 :::
 
-::: tip
+::: tip CID conversion for subdomains
+
+If you have content identified by an older CID v0, there are easy ways to safely represent it as CID v1 for use in subdomains and other case-insensitive contexts.
+
+#### Automatic – leverage the gateway in go-ipfs
+
+**TL;DR:** using subdomain gateway as a drop-in replacement for a path one removes the need for manual CID conversion.
+
+Request for a content path sent to the gateway domain will return HTTP 301 redirect to a correct subdomain version, taking care of any necessary encoding conversion, if needed:
+
+    https://<gateway-host>.tld/ipfs/<cid> -> https://<cidv1>.ipfs.<gateway-host>.tld/
+
+To illustrate, opening the CID v0 resource at [`https://dweb.link/ipfs/QmT5NvUtoM5nWFfrQdVrFtvGfKFmG7AHE8P34isapyhCxX/wiki/Mars.html`](https://dweb.link/ipfs/QmT5NvUtoM5nWFfrQdVrFtvGfKFmG7AHE8P34isapyhCxX/wiki/Mars.html)
+returns a redirect to a CID v1 representation at [`https://bafybeicgmdpvw4duutrmdxl4a7gc52sxyuk7nz5gby77afwdteh3jc5bqa.ipfs.dweb.link/wiki/Mars.html`](https://bafybeicgmdpvw4duutrmdxl4a7gc52sxyuk7nz5gby77afwdteh3jc5bqa.ipfs.dweb.link/wiki/Mars.html).
+
+Gateway takes care of converting CID to case-insensitive encoding.
+The multihash in CID v1 is the same as in original CID v0.
+
+#### Manual – use CID website or commandline
 
-To convert CID to Base32 use [cid.ipfs.io](https://cid.ipfs.io) or a command line:
+One can also do the conversion manually.
+To convert CID to Base32 ([RFC4648](https://tools.ietf.org/html/rfc4648#section-6), no padding) use [cid.ipfs.io](https://cid.ipfs.io) or a command line:
 
 ```shell
 > ipfs cid base32 QmbWqxBEKC3P8tqsKc98xmWNzrzDtRLMiMPL8wBuTGsMnR
 bafybeigdyrzt5sfp7udm7hu76uh7y26nf3efuylqabf3oclgtqy55fbzdi
 ```
 
-[PeerIDs can be represented as CID with `libp2p-key` multicodec](https://github.com/libp2p/specs/blob/master/RFC/0001-text-peerid-cid.md):
+PeerIDs can be represented as [CID with `libp2p-key` multicodec](https://github.com/libp2p/specs/blob/master/RFC/0001-text-peerid-cid.md).
+Base36 is suggested as a safer default for longer keys:
 
 ```shell
-> ipfs cid format -v 1 -b base32 --codec libp2p-key QmNnooDu7bfjPFoTZYxMNLWUQJyrVwtbZg5gBMjTezGAJN
-bafzbeiagwnqiviaae5aet2zivwhhsorg75x2wka2pu55o7grr23ulx5kxm
+> ipfs cid format -v 1 -b base36 --codec libp2p-key QmNnooDu7bfjPFoTZYxMNLWUQJyrVwtbZg5gBMjTezGAJN
+k2k4r8jl0yz8qjgqbmc2cdu5hkqek5rj6flgnlkyywynci20j0iuyfuj
```

Comment:
This only works on RSA keys, Ed25519 keys fail here. Ed25519 keys aren't super widely used right now, but as go-ipfs is switching to using them by default and PeerIDs will still be encoded as base58 by default we might want to include.

`ipfs id --format "<id>" --peerid-base base36` or using `ipfs key list -l` (we unfortunately haven't added support yet for `ipfs key list -l mykey`, but hopefully that'll make it into either the final release or 0.8.0)

Reply:
Eek. This edge case again :( 

Documenting this won't help much in practice: people will have pretty bad UX when `ipfs cid format` fails on ED25519 without an obvious reason, only a subset will read the docs and understand whats happening.

AFAIK `ipfs id` is the only place with non-CID PeerID when Ed25519 is used. 
`ipfs key list -l` prints Base36 version for `self` already.

@aschmahmann : how do you feel about `ipfs id` returning CIDv1b36 when PeerID is an identity multihash? It will close the UX gap.

I know its special-casing, but imo its worth it: will save us a lot of headache and time on user support in the following months.

Reply:
The issue is that the libp2p ecosystem is unprepared for peerIDs as CIDs and aside from this most recent go-libp2p (and go-ipfs) release nobody can process multiaddresses like `/p2p/k51abc...` which means the "normal" user flow of `ipfs swarm connect {copy-paste ipv4 part of multiaddress}/p2p/{copy-paste peerID} won't work on older nodes.

How commonly/why would people be converting the representations of peerIDs anyway? In my understanding using peerIDs in subdomains is not yet a commonly used feature. As a result my inclination is to just let this edge case be a bit miserable for a while and fix it later.

Some discussion on options around dealing with this are https://github.com/libp2p/go-libp2p/issues/976.

@Stebalien WDYT?

Reply:
Ack, I think p2p tunel is still experimental, so the only real life use will be IPNS on subdomains.
We already encourage people to publish to IPNS using a key other than `self`, so you are right, should be fine to leave it as-is.

Reply:
SGTM. Something that might help here is if we implement `ipfs key list -l {keyID}` since that would let us get the CID representation of the self key easily for publishing i.e. `ipfs key list -l self`.

Reply:
Good idea. I'll update this section with examples from https://github.com/ipfs/blog/pull/478